src/main/java/org/openrewrite/java/migrate/javax/HttpSessionInvalidate.java

```diff
@@ -0,0 +1,102 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.javax;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class HttpSessionInvalidate extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Updates `javax.servlet.http.HttpSession invalidate()` to `javax.servlet.http.HttpServletRequest logout()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Updates `javax.servlet.http.HttpSession invalidate()` to `javax.servlet.http.HttpServletRequest logout()`";
+    }
+
+    private final MethodMatcher INVALIDATE_METHOD_PATTERN = new MethodMatcher("javax.servlet.http.HttpSession invalidate()", false);
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new HttpSessionVisitor();
+    }
+
+    public class HttpSessionVisitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            if (INVALIDATE_METHOD_PATTERN.matches(method)) {
+                // Get index of param for HttpServletRequest, from the encapsulating method declaration TODO: would like to make this cleaner...
+                J.MethodDeclaration parentMethod = getCursor().dropParentUntil(parent -> parent instanceof J.MethodDeclaration).getValue();
+                Integer servletReqParamIndex = getServletRequestIndex(parentMethod);
+
+                // failed to find HttpServletRequest from parent MethodDeclaration
+                if (servletReqParamIndex == null) {
+                    return method;
+                }
+
+                // Get the HttpServletRequest param
+                J.VariableDeclarations httpServletRequestDeclaration = (J.VariableDeclarations) parentMethod.getParameters().get(servletReqParamIndex);
+
+                // Replace HttpSession.invalidate() with HttpServletRequest.logout()
+                final JavaTemplate logoutTemplate =
+                        JavaTemplate.builder("#{any(javax.servlet.http.HttpServletRequest)}.logout()")
+                                .imports("javax.servlet.http.HttpServletRequest")
+                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "javax.servlet-3.0"))
```

Comment:
The `javax.servlet-3.0` jar is only in `src/test/resources/META-INF/classpath`; Perhaps we should move that to `src/main` if we now also need it outside tests.